src/main/java/org/apache/sling/resourceresolver/impl/ResourceResolverImpl.java

```diff
@@ -1117,4 +1142,68 @@ public Resource move(final String srcAbsPath, final String destAbsPath) throws P
         }
         return rsrc;
     }
+
+
+
+    // A very simple tupel implementation which can be used as key in any map
+    public class String3Tupel {
+
+        @Override
+        public int hashCode() {
+            final int prime = 31;
+            int result = 1;
+            result = prime * result + getEnclosingInstance().hashCode();
```

Comment:
Is this necessary? A tuple3 does not escape the scope of a single resource resolver, right?

Reply:
I am still missing the point of this field, seems it's only used in hashcode/equals - is it really necessary?